cni/pkg/util/podutil.go

```diff
@@ -51,19 +51,25 @@ var annotationRemovePatch = []byte(fmt.Sprintf(
 
 // PodRedirectionEnabled determines if a pod should or should not be configured
 // to have traffic redirected thru the node proxy.
-func PodRedirectionEnabled(namespace *corev1.Namespace, pod *corev1.Pod) bool {
-	if !(namespace.GetLabels()[label.IoIstioDataplaneMode.Name] == constants.DataplaneModeAmbient ||
-		pod.GetLabels()[label.IoIstioDataplaneMode.Name] == constants.DataplaneModeAmbient) {
+func PodRedirectionEnabled(namespace *corev1.Namespace, pod *corev1.Pod, enablementSelector labels.Selector) bool {
+	nsMatches := enablementSelector.Matches(labels.Set(namespace.GetLabels()))
+	podMatches := enablementSelector.Matches(labels.Set(pod.GetLabels()))
+	if !(nsMatches || podMatches) {
 		// Neither namespace nor pod has ambient mode enabled
 		return false
 	}
 	if podHasSidecar(pod) {
 		// Ztunnel and sidecar for a single pod is currently not supported; opt out.
 		return false
 	}
-	if pod.GetLabels()[label.IoIstioDataplaneMode.Name] == constants.DataplaneModeNone {
-		// Pod explicitly asked to not have ambient redirection enabled
-		return false
+	// TODO (therealmitchconnors): how to accomplish this with the enablement selector?
```

Comment:
My other comment above will enable. I could have sworn I had some examples written somewhere of configs that would do this but I am not seeing it